src/routes/getSkipSegments.js

```diff
@@ -8,271 +8,179 @@ var privateDB = databases.privateDB;
-  let weightSums = [];
-
-  for (let i = 0; i < choiceGroups.length; i++) {
-      //find weight sums for this group
-      weightSums.push(0);
-      for (let j = 0; j < choiceGroups[i].length; j++) {
-          //only if it is a positive vote, otherwise it is probably just a sponsor time with slightly wrong time
-          if (weights[choiceGroups[i][j]] > 0) {
-              weightSums[weightSums.length - 1] += weights[choiceGroups[i][j]];
-          }
-      }
-
-      //create a random choice for this group
-      let randomChoice = getWeightedRandomChoice(choiceGroups[i], weights, 1)
-      finalChoices.push(randomChoice.finalChoices);
-
-      for (let j = 0; j < randomChoice.choicesDealtWith.length; j++) {
-          choicesDealtWith.push(randomChoice.choicesDealtWith[j])
-      }
-  }
-
-  return {
-      finalChoices: finalChoices,
-      choicesDealtWith: choicesDealtWith,
-      weightSums: weightSums
-  };
-}
-
-//gets a weighted random choice from the indexes array based on the weights.
-//amountOfChoices speicifies the amount of choices to return, 1 or more.
+//gets a weighted random choice from the choices array based on their `votes` property.
+//amountOfChoices specifies the maximum amount of choices to return, 1 or more.
 //choices are unique
-function getWeightedRandomChoice(choices, weights, amountOfChoices) {
-  if (amountOfChoices > choices.length) {
-      //not possible, since all choices must be unique
-      return null;
+function getWeightedRandomChoice(choices, amountOfChoices) {
+  //trivial case: no need to go through the whole process
+  if (amountOfChoices >= choices.length) {
+    return choices;
   }
 
-  let finalChoices = [];
-  let choicesDealtWith = [];
+  //assign a weight to each choice
+  let totalWeight = 0;
+  choices = choices.map(choice => {
+    //multiplying by 10 makes around 13 votes the point where it the votes start not mattering as much (10 + 3)
+    //The 3 makes -2 the minimum votes before being ignored completely
+    //https://www.desmos.com/calculator/ljftxolg9j
+    //this can be changed if this system increases in popularity.
+    const weight = Math.sqrt((choice.votes + 3) * 10);
+    totalWeight += weight;
 
-  let sqrtWeightsList = [];
-  //the total of all the weights run through the cutom sqrt function
-  let totalSqrtWeights = 0;
-  for (let j = 0; j < choices.length; j++) {
-      //multiplying by 10 makes around 13 votes the point where it the votes start not mattering as much (10 + 3)
-      //The 3 makes -2 the minimum votes before being ignored completely
-      //https://www.desmos.com/calculator/ljftxolg9j
-      //this can be changed if this system increases in popularity.
-      let sqrtVote = Math.sqrt((weights[choices[j]] + 3) * 10);
-      sqrtWeightsList.push(sqrtVote)
-      totalSqrtWeights += sqrtVote;
-
-      //this index has now been deat with
-      choicesDealtWith.push(choices[j]);
-  }
+    return { ...choice, weight };
```

Comment:
Why is choice turned into an array?

Reply:
It's not turned into an array, it [spreads](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Syntax) `choice`'s properties in a new object and adds the `weight` property to it.